sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -195,19 +231,26 @@ case class Hour(child: Expression) extends UnaryExpression with ImplicitCastInpu
       > SELECT _FUNC_('2009-07-30 12:58:59');
        58
   """)
-case class Minute(child: Expression) extends UnaryExpression with ImplicitCastInputTypes {
+case class Minute(child: Expression, timeZoneId: Option[String] = None)
```

Comment:
No, `Minute` / `Second` should also be timezone-aware because `TimeZone` can have millisecond offset. Actually, there are some timezones which have offset of some hours and 30 minutes. There is no timezone which has non-zero second offset for now, though.